Tsconfig, Webpack, Babel, StyleLint 적용

Do-ho opened the pull request:
<!-- 작업 내용이 무엇인지 적어주세요 -->
<!-- 작업이 여러개라면 ###을 이용해 문단 구분을 해주세요 -->
<!-- 관련 이슈가 있다면 추가해주세요 -->

## 공통
- alias는 src에 대해서 `@`로 처리했습니다.


## TSConfig 기본 설정

### 상세 작업 내용

- 기본적인 옵션들은 옛날 tsConfig에서 가져왔습니다.
- 옵션 설명입니다. [참조](https://typescript-kr.github.io/pages/compiler-options.html)

compilerOptions | 설명
-- | --
target | ECMAScript 대상 버전 지정입니다. 최신으로 하기 위해 ESNext로 지정
allowJS | JavaScript 파일의 컴파일 허용
skipLibCheck | .d.ts 타입의 검사 건너뜀
esModuleInterop | ES6 모듈 사양 준수하여 CommonJS 모듈을 가져올 수 있게 함
strict | 모든 엄격한 타입 검사 옵션 활성화
forceConsistentCasingInFileNames | 경로 제대로 적게하기 ex) 대소문자
module | 모듈 코드 생성 지정 -> ESNext로 지정
moduleResolution | 모듈 해석 방법 결정, Node 스타일 적용
resolveJsonModule | json 확장자로 import된 모듈 포함
isolatedModules | typescript에 빈페이지가 없도록 함
noEmit | 출력을 내보내지 않음
jsx | tsx 파일에서 JSX 지원
noImplicitAny | any 타입 선언에 대한 오류 반환
noImplicitThis | any 타입으로 표현한 this에 오류 반환
strictNullChecks | 엄격한 null 검사

## ESLint, Prettier 설정 변경

### 상세 작업 내용
- [x] **eslint-import-resolver-typescript** 설치
- typescript에 import 시키는 여러 모듈에 연동을 위한 설치입니다.
- [x] `json`에서 `js`로 파일 확장자명을 공통화 시켰습니다.

## Babel 설정

### 상세 작업 내용
- [x] **@babel/preset-env** : ECMAScript2015+를 변환하기위해 사용
  - "last 2 versions", ">= 5% in KR"의 속성을 적용했습니다.
- [x] **@babel/preset-react** : react를 변환하기 위한 프리셋
- [x] **@babel/preset-typescript** : typescript를 변환하기 위한 프리셋

## Webpack 설정

### 상세 작업 내용
- [x] **babel-loader** : babel을 적용하기 위한 사용
- [x] **ts-loader** : typescript를 적용하기 위한 사용
- 여기서 transpileOnly를 true로 해주면서 타입체크를 수행하지 않는데 fork-ts-checker-webpack-plugin을 이용해 별도의 프로세스에서 타입체크를 하게 했습니다.
- 타입스크립트의 안정성 및 빠른 빌드 속도를 위한 결정이었습니다.
- [x] **url-loader**
- 향후 다양한 사진을 사용할 수 있으므로 미리 loader에 추가시켰습니다.
- [x] **mini-css-extract-plugin**
- `style-loader`를 사용하지 않고 해당 패키지를 사용한 이유는 따로 `style.css` 파일로 빼서 가져오기 위함입니다.
- [x] **css-loader, sass-loader**
- `scss` 파일을 적용하기 위해 사용

webpack 파일 위쪽에 require 문을 사용하기 위해 해당 파일에서만 off 시켰습니다.

## StyleLint 설정
기존에는 sass-lint를 사용하려했으나 auto-save 기능이 없는 것 같아 styleLint로 급하게 바꿨습니다.

### 상세 작업 내용

- [x] **styleLint Library** 설치

| Library                   | Content                                                      |
| ------------------------- | ------------------------------------------------------------ |
| stylelint-config-standard | stylelint의 유명한 기업이나 스타일 가이드 라인에서 쓰이는 룰을 적용하기 위해 도입 |
| stylelint-order           | stylelint의 css 속성 순서를 결정해주기 위해 도입             |

- [x] **styleLint** 적용

Rule | Content
-- | --
At-rule-no-unknown <br /> scss/at-rule-no-unknown | 알려지지 않은 문법을 무시함 (scss 같은 경우 @를 사용하기 위해 null로 표시)



### 공유
- extension으로 stylelint를 설치해주세요 ㅎㅎ
- `./vscode/settings.json`에 해당 내용을 추가해주셔야 auto-save가 발생합니다.
```
{
  "editor.codeActionsOnSave": {
     "source.fixAll.eslint": true,
     "source.fixAll.stylelint": true
   },
   "stylelint.enable": true,
 }

```

## 관련 이슈
closed #4 
closed #6

Do-ho committed:
chore: 초기 tsconfig 설정파일 추가

Do-ho committed:
refactor: prettier, eslint 확장자 통일

- 다른 설정파일과 확장자를 통일하기 위해서 변경

Do-ho committed:
chore: Babel 관련 패키지 설치

Do-ho committed:
feat: babel 설정 파일 작성

Do-ho committed:
chore: webpack 관련 패키지 설치

Do-ho committed:
feat: webpack 관련 CLI 설정

Do-ho committed:
feat: webpack 설정 파일 작성

Do-ho committed:
feat: scss + css modules 적용을 위한 임시 파일 생성

Do-ho committed:
feat: 전역 scss 파일에 대한 declare 설정

- Typescript 검사에서 scss를 import 하기 위해서 declare 하였음
- typescript 상에서 ts(2307) 에 대한 에러 내용

Do-ho committed:
chore: stylelint 관련 패키지 설치

Do-ho committed:
feat: stylelint 설정 파일 작성

Do-ho committed:
chore: build와 vscode 개별 설정을 위한 gitignore 추가

Do-ho committed:
feat: eslint에 tsconfig 연결

Do-ho committed:
chore: eslint와 tsconfig 연동을 위한 패키지 설치

jinhyukoo reviewed (commented):
#7 에 남긴 내용 반영 잘 해주셨네요. 감사합니다. 추후 mode에 따른 분기만 추가되면 될 것 같습니다. 그런데 mode 이야기가 나와서 그런데 mode값 입력 안 해주면 production이 default값이 아니었나요? mode: 'development'를 추가해줘야 할 것 같네요. stylelint 쪽은 저도 정확한 지식이 없어서 피드백을 못하겠네요. ㅠㅠㅠ

+++++++내용추가++++++++

> By setting the mode parameter to either development, production or none, you can enable webpack's built-in optimizations that correspond to each environment. The default value is production.

웹팩 문서에서 확인해봤더니 기본값이 production이네요. 우선은 개발 모드로 설정하는게 좋을 듯 싶습니다.

wooojini commented on webpack.config.js:
```diff
@@ -0,0 +1,67 @@
+      template: "./public/index.html"
+    }),
+    new MiniCssExtractPlugin({ filename: "style.css" }),
+    new ForkTsCheckerWebpackPlugin()
```
타입 체크를 번들러에서 돌리지 않고, `tsc 커맨드`를 이용해서 `husky`에서 체크하기로 하지 않았나요?

Do-ho replied:
음.. 혹시 처음 번들링할 때 타입 체크를 하지 않는 이유가 있을까요? 속도 측면에서의 문제일까요

wooojini replied:
음.. 프로젝트가 커지면 빌드 속도가 오래걸리기도 하고, 
매번 개발하면서 빌드할 때마다 타입 체크를 할 필요가 있을까 싶어서요 🧐

런타임 전 타입은 에디터에서 체크해주고, 
실제 컴파일 체크는 커밋 할 때 훅에서 체크하면 충분해보여서요 :)

Do-ho replied:
음... 아무래도 development와 production을 나눠야하는 이유가 여기서도 보이네요.
실제 배포 환경에서는 type 검사를 거쳐서 배포하는게 더 좋다고 생각이 들어서요.

@jinhyukoo

이번 PR에서 환경 분리도 같이 진행하겠습니다 :)

wooojini replied:
@Do-ho 
흠 타입 체크를 한다고해도 CI 체크시 jest test 파일 검증하는 것과 동일하게 CLI 커맨드로 테스트 해도 될 것 같은데 프로덕션 파일 설정로 분리할 필요가 있을까요?

Do-ho replied:
CI를 생각을 못했네요..! 그렇다고 하면 type check 로직에 대한 빈틈이 생각하기에 없는 것 같습니다. typechecker도 걷어내도록 하겠습니다 :)

wooojini commented on .stylelintrc.js:
```diff
@@ -0,0 +1,79 @@
+  rules: {
+    "at-rule-no-unknown": null,
+    "scss/at-rule-no-unknown": true,
+    "order/properties-order" :[
```
오! CSS 프로퍼티 순서 체킹이 있네요! 요거 맨날 찾아보면서 했는데 좋네요 :)

wooojini reviewed (commented):
고생하셨습니다 🙏
stylelint 처음 써보는데 auto-save 좋네요 :)
몇 가지 코멘트 내용 남겼습니다.

wooojini commented on global.d.ts:
```diff
@@ -0,0 +1,4 @@
+declare module "*.module.scss" {
+  const classes: { [key: string]: string };
```
classes를 선언하신 이유가 있나요??

Do-ho replied:
이 부분 또한 css-loader에 modules: true를 하지 않아 추가한 코드였습니다.. ㅠㅠ

Do-ho commented:
@jinhyukoo 

> 추후 mode에 따른 분기만 추가되면 될 것 같습니다. 그런데 mode 이야기가 나와서 그런데 mode값 입력 안 해주면 production이 default값이 아니었나요? mode: 'development'를 추가해줘야 할 것 같네요. stylelint 쪽은 저도 정확한 지식이 없어서 피드백을 못하겠네요. ㅠㅠㅠ

webpack cli 관련 커밋 보시면 mode에 대해서 설정해주는 부분이 있습니다 :) [참조](https://webpack.js.org/configuration/mode/#root)

Do-ho commented on .eslintrc.js:
```diff
@@ -0,0 +1,53 @@
+    sourceType: "module"
+  },
+  plugins: ["react", "@typescript-eslint", "prettier"],
+  settings: {
```
앗 `git rebase -i hash`하고 commit해서 추가했었는데 뭔가 꼬인 것 같네요.. 
```
"import/resolver": {
      typescript: "./tsconfig.json",
      node: {
        paths: ["src"],
        extensions: [".js", ".jsx", ".ts", ".tsx"]
      }
    }
```

여기서 `"./tsconfig.json`이 추가된 커밋이였습니다

Do-ho committed:
fix: babel brower version 변경

Do-ho commented on webpack.config.js:
```diff
@@ -0,0 +1,67 @@
+        use: [
+          "babel-loader",
+          {
+            loader: "ts-loader",
```
제가 알기로는 babel로 해결되지 않는 것을 ts-loader를 통해 해결해주는 무엇인가가 있다고 해서 같이 썼었는데 확인을 해볼게요 ㅎㅎ

wooojini replied:
제가 궁금했던 점은 `babel`과 `ts-loader`를 둘 다 사용하신 이유였는데요 ㅎ 
`babel`로 트랜스파일을 거쳤는데 `ts-loader`를 한번 더 거치는 이유가 있는지 의문이여서요 :)

babel 7부터 공식적으로 [typescript](https://babeljs.io/docs/en/babel-preset-typescript)를 지원하고 있어서 `ts-loader`를 사용해야하는 특별한 이유가 없다면 `babel`만 사용해도 충분하다는 생각입니다.

wooojini replied:
`ts-loader`는 [HMR 지원](https://github.com/TypeStrong/ts-loader#hot-module-replacement)을 하지 않네요.. ㅠ

Do-ho replied:
오... 감사합니다!

보셨는 지는 모르겠지만 [김정환님 블로그](https://jeonghwan-kim.github.io/dev/2021/03/08/babel-typescript.html)에는 이런 식으로도 이야기하네요.

babel loader에서는 생성자함수를 undefined로 초기화하는 코드가 생성된다고 합니다. 아무래도 ts-loader를 통해 생성된 결과물과 babel-loader를 통해서 생성된 결과가 조금 다른 것 같습니다.

> 필드 선언(Field declaration)은 아직 실험적 기능(stage 3)이기 때문에 이를 사용하려면 바벨의 @babel/plugin-proposal-class-properties 플러그인을 추가해야한다.

아직은 필드 선언에 대해서는 실험적 기능이긴한데 프로젝트에 어떤 영향을 미칠 지는 잘 모르겠네요.
우선 babel만으로도 typescript를 빌드 할 수 있으니 ts-loader는 걷어내겠습니다

Do-ho replied:
그런데 궁금한게 babel에서도 타입 검사에 대한 로직을 거친다면 ts-loader랑 많은 차이가 날까요?

wooojini replied:
@Do-ho 
오! 그렇다면 더욱이 ts-loader를 걷어내야겠네요!
위 블로그는 저도 참조했는데요. 정환님 의견대로 JS 스펙에 추가되지 않은 것은 당연히 preset을 추가하고 유지/관리하는 것이 맞다고 생각합니다 :)
무엇보다 HMR 지원하지 않으면 개발할 때 너무 불편합니다....ㅠ

Do-ho commented on webpack.config.js:
```diff
@@ -0,0 +1,67 @@
+      },
+      {
+        test: /\.s[ac]ss$/,
+        use: [MiniCssExtractPlugin.loader, "css-loader", "sass-loader"]
```
위의 커맨트랑 같이 해결하면서 왜 저게 안되는지 확인했습니다 ㅠㅠ
css에 modules를 true로 바꿔야했군요... 감사합니다.

Do-ho committed:
fix: css-loader에 modules 옵션 추가

- css-modules를 사용하기 위해서는 modules option 추가를 해야한다.
- 또한 기존에 해당 문제를 해결하기 위해 설정했던 옵션들을 모두 제거하였다.

Do-ho committed:
chore: assets.d.ts 파일 폴더 변경

- global.d.ts에서 assets.d.ts로 파일 이름 변경

Do-ho commented on babel.config.js:
```diff
@@ -0,0 +1,14 @@
+      "@babel/preset-env",
+      {
+        targets: {
+          browsers: ["last 2 versions", ">= 5% in KR"]
```
1449598 에서 변경했습니다 :)

Do-ho commented on global.d.ts:
```diff
@@ -0,0 +1,4 @@
+declare module "*.module.scss" {
```
a8d6936 에서 변경했습니다 :)

wooojini commented on babel.config.js:
```diff
@@ -0,0 +1,14 @@
+      }
+    ],
+    "@babel/react",
+    "@babel/typescript"
```
요기 바벨 프리셋 설정 방식이 바뀌었는데요. 확인해주세용 :)
- [@babel/preset-react](https://babeljs.io/docs/en/babel-preset-react#with-a-configuration-file-recommended)
- [@babel/preset-typescript](https://babeljs.io/docs/en/babel-preset-typescript#usage)

Do-ho committed:
chore: ts-loader 삭제

- babel 7이후 typescript를 지원하면서 typescript에 대한 파일 변환을 @babel/preset-typescript에서 가능해졌으므로 제거

Do-ho committed:
fix: webpack에서 ts-loader 제거

- ts-loader를 제거함에따라 loader 이용 부분 삭제

Do-ho committed:
fix: babel preset 설정 변경

Do-ho committed:
chore: webpack-merge 설치

Do-ho committed:
feat: webpack config 파일 분리

- webpack 코드에 대한 관리를 편하게 하기 위해 production, develop 모드를 분리

Do-ho committed:
refactor: webpack cli 변경

Do-ho committed:
chore: fork-ts-checker-webpack-plugin 삭제

- type checker는 tsc와 husky 그리고 ci를 통해 할 것이므로 빌드 환경 최적화를 위한 삭제

Do-ho committed:
fix: webpack fork-ts-checker-webpack-plugin 코드 삭제

-  fork-ts-checker-webpack-plugin 라이브러리가 삭제됨에 따라 코드 변경

wooojini commented on package.json:
```diff
@@ -5,8 +5,8 @@
     "test": "echo \"Error: no test specified\" && exit 1",
-    "start": "webpack-dev-server --mode development --open --hot",
-    "build": "webpack --mode production"
+    "start": "webpack serve --open --hot --config webpack.config.dev.js",
```
@Do-ho 
@jinhyukoo 

환경설정 고생하셨습니다 🙏
- `--open` 옵션은 개인적으로 개발시에 빌드를 새로해야하는 경우가 있는데 그때마다 브라우저 새탭이 열려서  조금 불편한데요. 다들 어떻게 생각하시나요? ㅎ
- `--hot` 옵션도 devServer 설정 파일에 추가해서 관리하면 좋을 것 같네요 :) ([devServer.hot](https://webpack.js.org/configuration/dev-server/#devserverhot))

jinhyukoo replied:
@wooojini 

- 저는 개인적으로 open 옵션이 있으면 편하긴 한데요. 많이 불편하시면 빼셔도 상관없습니다!

Do-ho replied:
@wooojini 저는 open 옵션 좋아하는게 다른 프로젝트 개발할 때마다 포트 바꿔줘야해서 조금 불편하더라고요 hot 옵션은 옮기도록 하겠습니다

wooojini replied:
@Do-ho @jinhyukoo 

두 분이 그렇다면 `hot` 옵션을 사용하는 것으로 가시죵 :)

wooojini commented on webpack.config.dev.js:
```diff
@@ -0,0 +1,13 @@
+  devServer: {
+    historyApiFallback: true,
+    port: 3000
+  }
```
develop, production 나눈김에 `css 소스맵`도 적용하면 좋을 것 같네요 :)
찾아보니 `MiniCssExtractPlugin`는 소스맵 적용이 안되는 것 같아서 개발 모드에서는 `style-loader`, 프로덕션 모드에서는 `MiniCssExtractPlugin`을 사용하면 될 것 같습니다.

아래는 테스트한 결과입니다 :)

- 적용 전
<img width="558" alt="스크린샷 2021-10-02 오후 11 52 20" src="https://user-images.githubusercontent.com/32856129/135722697-44400b2f-3ed6-4050-aacc-b0fdfa1b44bd.png">

- 적용 후
<img width="624" alt="스크린샷 2021-10-03 오전 12 13 54" src="https://user-images.githubusercontent.com/32856129/135722725-b92c2d1f-5906-455a-8e01-3d567374c3b3.png">

- 설정 참고
```
// webpack.config.dev.js
module: {
    rules: [
      {
        test: /\.s[ac]ss$/,
        use: [
          "style-loader",
          { loader: "css-loader", options: { modules: { localIdentName: "[local]__[hash:base64:5]" } } },
          { loader: "sass-loader", options: { sourceMap: true } }
        ]
      }
    ]
  }
```

```
// webpack.config.prod.js
module: {
    rules: [
      {
        test: /\.s[ac]ss$/,
        use: [MiniCssExtractPlugin.loader, { loader: "css-loader", options: { modules: true } }, "sass-loader"]
      }
    ]
  },
  plugins: [new MiniCssExtractPlugin({ filename: "style.css" })]
```

Do-ho replied:
오 감사합니다. 혹시
localIdentName에서 path랑 name을 같이 쓰는 것은 어떻게 생각하세요?
`[path][name]__[local]_[hash:base64:5]`

wooojini replied:
음.. 개인적으로는 class만 있어도 충분하다는 생각입니다.
컴포넌트 단위로 css 작업을 하기로 하였고, 경로는 코드상에서 충분히 확인할 수 있어서요.

DOM에서는 클래스 이름을 명확하게 확인할 수 있는게 좋을 것 같다는 생각인데요.
경로 depth가 깊어지거나 class 이름이 길어지거나 하면 오히려 불편할 것 같아서요 :)

@jinhyukoo 
요건 진혁님 의견도 들어보죠 ㅎㅎ

- 경로, 파일이름 적용시 
<img width="162" alt="스크린샷 2021-10-04 오후 5 38 07" src="https://user-images.githubusercontent.com/32856129/135820159-6aacbf1f-f3a9-4483-9e54-8bcaf24e8c4d.png">

jinhyukoo replied:
@Do-ho

저도 우진님과 같은 의견인데요. 혹시 경로와 파일명을 추가하고 싶으신 이유가 어떤건지 여쭤봐도 될까요? 

P.S 제가 핸드폰으로 보다보니 다른 커멘트를 못봤는데 혹시 다른 곳에 이유를 남겨두셨다면 죄송해욤...

wooojini replied:
@Do-ho 

권장하기는 하나 조금 공감은 안가는데요 :)

**소스맵에 경로를 넣게 된다면 depth가 깊어졌을 때 경로 찾기가 더 수월하지 않을까**
   - 컴포넌트 depth가 깊어지고 마크업 구조가 복잡한 경우, 비슷한 네이밍과 이름이 길어지는 컴포넌트들이 생긴다면 DOM 트리에서 class 이름을 직관적으로 파악하기 힘들 것 같아서요.
   - 이런 상황에서 class 이름을 바꾸거나, 추가하거나 한다면 제대로 바뀌었는지 직관적으로 파악이 어려울 것 같네요...
 
**소스를 통해서 확인하는 것도 좋지만 제 경험상으로 한번에 경로가 읽히는 경우는 잘 없다**
   - 소스상에서 경로를 보고 찾아간다기보다는 에디터에서 바로 해당 파일을 찾아갈 수 있고, 사이드 탭에 폴더 구조도 해당 경로에 맞게 열어주어서 개인적으로는 불편함이 없었는데요. 
   - 오히려 소스상에서도 파악하기 힘들정도의 depth라면 깊은 depth 구조일텐데 class에 해당 경로가 그대로 추가되어 DOM 트리에서 파악한다고해도 마찬가지일 것 같습니다. 

**다른 사람이 작성한 코드의 css를 변경하고 싶다면 구조를 다시 한 번 보고 찾아갈 것**
   - 다른 사람이 작성한 코드라면 더욱이 컴포넌트 depth 구조나 네이밍도 익숙치 않아서 우선 소스부터 확인해야할 것 같은데요. DOM 트리에서 class 이름이 바로 파악되면 소스에서 바로 확인할 수 있고 에디터에서 해당 파일을 찾아가서 수정하기 수월할 것 같습니다.

Do-ho replied:
소스맵의 사용 용도에 따라서 결정을 하는게 맞을 것 같네요.

에디터를 통한 검색은 생각하지 못했네요.
우진님이 이야기해주신 것처럼 에디터에서 클래스명을 통한 검색이 쉽게 가능할 것 같습니다.

또한 저도 경로는 heavy한 정보같습니다.

### 다만 filename의 추가에 대해서는 이야기해보고 싶은데요.

- 에디터 서칭할 때 className을 복붙해와 검색하면
클래스명의 중복으로 에디터 서치 결과가 여러 개가 나올 수 있다고 생각합니다.
그래서 힌트처럼 filename도 같이 가져가 보는 것도 괜찮을 것 같습니다.
에디터 상에서 filename을 통해서 먼저 찾아볼 수 있을테니까요.

- 웬만하면 컨벤션 상 파일 이름이 컴포넌트의 이름이 될 것이라고 생각합니다.
그랬을 때 소스상에서 컴포넌트의 이름을 인식할 수도 있겠네요.

### 정리해보면 3가지의 케이스로 나뉠 것 같습니다.

1. `[local]__[hash:base64:5]`
2. `[local]__[hash:base64:5]__[name]`
3. `[name]__[local]__[hash:base64:5]`

3번은 약간 bem 표기법처럼 보일 수도 있겠네요.

한 번 생각해보시고 영 아니다 싶으면 1번으로 가겠습니다.

@wooojini @jinhyukoo

wooojini replied:
@Do-ho @jinhyukoo 

넵 클래스가 여러개 달리거나하는 복잡한 상황도 고려해야한다는 생각이 떠나질 않아서 제 의견은 1번인데용 :) 
하지만 두 분이 파일이름을 추가하는 것에 동의하신다면 저도 따라가겠습니다 👌

jinhyukoo replied:
@wooojini @Do-ho 

제가 확인이 많이 늦었네요. 죄송합니다. 
우선 두 분다 좋은 의견 내주셔서 감사드려요. 

단순히 파일을 찾아가기 위해 fileName을 추가하는 거라면 굳이 각 클래스 명 앞에 prefix로 fileName을 붙이는게 아니고 조금 번거롭지만 class의 값으로 fileName을 추가해버리는게 나은 선택이 아닌가 싶습니다. 우진님 말씀처럼 클래스가 여러개 있을 경우 불필요하게 fileName이 반복될 것 같아서요!

Do-ho replied:
@jinhyukoo @wooojini 

넵 알겠습니다 ㅎㅎ 그러면 `[local]__[hash:base64:5]`로 통일해서 커밋 다시 남기겠습니다 :)

Do-ho committed:
chore: development hot 옵션 CLI에서 Webpack 설정으로 변경

Do-ho committed:
chore: style-loader 설치

Do-ho committed:
feat: webpack development 모드 style-loader로 변경

- 더 빠른 dev-server를 위해 MiniCssExtractPlugin.loader에서 library를 바꿈

Do-ho committed:
fix: webpack 경로 문제 해결

- development와 production의 publicPath가 상이한 것을 확인하였음.
- 따라서 auto로 바꿔서 공용 경로로 결정해 자동으로 처리

Do-ho commented:
42da6fe

커밋에 대해서 webpack에서 development와 production에서 경로가 이상하게 나와 css와 js 파일을 가져오지 못하는 현상이 있었습니다.
해당 부분은 publicPath의 경로를 통해서 파일에 `/`가 붙거나 안붙거나로 나뉘어져 link 태그에 적용되는 것이 달랐습니다.

- development
 
![image](https://user-images.githubusercontent.com/33643752/135752800-d9bff37c-883c-4bf0-a05d-ff0f274de9ab.png)

- production
![image](https://user-images.githubusercontent.com/33643752/135752843-45666f70-5077-4088-bd65-fa38e6b7b78f.png)

production mode에서는 다음과 같은 오류를 냈습니다.
![image](https://user-images.githubusercontent.com/33643752/135752870-0be02f85-2e27-43c6-b7fd-5dc086b2f670.png)

결국 production과 development의 publicPath를 auto로 바꿔 공용 경로로 결정해서 자동으로 처리하도록 했습니다.

Do-ho committed:
refactor: webpack develop의 localIdentName 변경

- 관리자 도구를 통해 한눈에 구조를 파악하기 위해서 변경
- 레거시 코드는 컴포넌트의 depth가 깊어졌을 경우 path와 name을 보고 해당 파일을 찾아가기 힘들 것이다.
- css-loader 라이브러리에서 권장하는 설정이긴하나 관리자도구를 통해 클래스 이름을 더 빠르게 찾아가고 변경을 하기 위해 반영하였습니다.

wooojini reviewed (approved):
@Do-ho 
환경설정 작업 고생많으셨습니다 🙏🙏🙏

jinhyukoo reviewed (approved):
LGTM
